code/modules/organs/organ_alien.dm

```diff
@@ -0,0 +1,229 @@
+	name = "polyp segment"
+	parent_organ = "groin"
+
+/datum/organ/internal/diona/ligament
+	name = "anchoring ligament"
+	parent_organ = "groin"
+
+/datum/organ/internal/diona/node
+	name = "receptor node"
+	parent_organ = "head"
+	removed_type = /obj/item/organ/diona/node
+
+/datum/organ/internal/diona/nutrients
+	name = "nutrient vessel"
+	parent_organ = "chest"
+	removed_type = /obj/item/organ/diona/nutrients
+
+/obj/item/organ/diona
+	name = "diona nymph"
+	icon = 'icons/obj/objects.dmi'
+	icon_state = "nymph"
+	organ_tag = "special" // Turns into a nymph instantly, no transplanting possible.
+
+/obj/item/organ/diona/removed(var/mob/living/target,var/mob/living/user)
+
+	..()
+	var/mob/living/carbon/human/H = target
+	if(!istype(target))
+		del(src)
+
+	if(!H.internal_organs.len)
+		H.death()
+
+	//This is a terrible hack and I should be ashamed.
+	var/datum/seed/diona = seed_types["diona"]
+	if(!diona)
+		del(src)
+
+	spawn(1) // So it has time to be thrown about by the gib() proc.
+		var/mob/living/carbon/alien/diona/D = new(get_turf(src))
+		diona.request_player(D)
+		del(src)
+
+// These are different to the standard diona organs as they have a purpose in other
+// species (absorbing radiation and light respectively)
+/obj/item/organ/diona/nutrients
+	name = "nutrient vessel"
+	organ_tag = "nutrient vessel"
+	icon = 'icons/mob/alien.dmi'
+	icon_state = "claw"
+
+/obj/item/organ/diona/nutrients/removed()
+	return
+
+/obj/item/organ/diona/node
+	name = "receptor node"
+	organ_tag = "receptor node"
+	icon = 'icons/mob/alien.dmi'
+	icon_state = "claw"
+
+/obj/item/organ/diona/node/removed()
+	return
+
+//CORTICAL BORER ORGANS.
+/datum/organ/internal/borer
+	name = "cortical borer"
+	parent_organ = "head"
+	removed_type = /obj/item/organ/borer
+	vital = 1
+
+/datum/organ/internal/borer/process()
+
+	// Borer husks regenerate health, feel no pain, and are resistant to stuns and brainloss.
+	for(var/chem in list("tricordrazine","tramadol","hyperzine","alkysine"))
+		if(owner.reagents.get_reagent_amount(chem) < 3)
+			owner.reagents.add_reagent(chem, 5)
+
+	// They're also super gross and ooze ichor.
+	if(prob(5))
+		var/mob/living/carbon/human/H = owner
+		if(!istype(H))
+			return
+
+		var/datum/reagent/blood/B = locate(/datum/reagent/blood) in H.vessel.reagent_list
+		blood_splatter(H,B,1)
+		var/obj/effect/decal/cleanable/blood/splatter/goo = locate() in get_turf(owner)
+		if(goo)
+			goo.name = "husk ichor"
+			goo.desc = "It's thick and stinks of decay."
+			goo.basecolor = "#412464"
+			goo.update_icon()
+
+/obj/item/organ/borer
+	name = "cortical borer"
+	icon = 'icons/obj/objects.dmi'
+	icon_state = "borer"
+	organ_tag = "brain"
+	desc = "A disgusting space slug."
+
+/obj/item/organ/borer/removed(var/mob/living/target,var/mob/living/user)
+
+	..()
+
+	var/mob/living/simple_animal/borer/B = target.has_brain_worms()
+	if(B)
+		B.leave_host()
+		B.ckey = target.ckey
+
+	spawn(0)
+		del(src)
+
+//XENOMORPH ORGANS
```

Comment:
Mhm, this file currently isn't included in the compile list. I'm not too sure yet if we're going to use alien organs, because it'd be a load of effort to also port over Bay's organ improvements.